website/src/blogs/2018-12-14-openebs-snapshots-using-kubectl.md

```diff
@@ -0,0 +1,282 @@
+    metadata:
+      name: snapshot-promoter
+    provisioner: volumesnapshot.external-storage.k8s.io/snapshot-promoter
+
+We can now create a PersistentVolumeClaim that will use the StorageClass to dynamically provision a PersistentVolume that contains the info of our snapshot. The annotation field snapshot.alpha.kubernetes.io/snapshot refers to the VolumeSnapshot object. Snapshot-provisioner will use this resource to obtain the information it needs to perform the snapshot restore. After this, snapshot-provisioner will provision a PersistentVolume containing the contents of the snapshot-demo snapshot
+
+    apiVersion: v1
+    kind: PersistentVolumeClaim
+    metadata:
+      name: demo-snap-vol-claim
+      annotations:
+        snapshot.alpha.kubernetes.io/snapshot: snapshot-demo
+    spec:
+      storageClassName: snapshot-promoter
+      accessModes: ReadWriteOnce
+      resources:
+        requests:
+          storage: 5Gi
+
+We can check the state of demo-snap-vol-claim to see if it is Bound or not. We can also view the snapshot-provisioner logs to verify that the snapshot was restored successfully.
+
+    $ kubectl logs snapshot-controller-66f7c56c4-ptjmb-c snapshot-provisioner 
+    ... 
+    ... 
+    I1104 11:59:10.563990       1 controller.go:813] volume "pvc-8eed96e4-c157-11e7-8910-42010a840164" for claim "default/demo-snap-vol-claim" created 
+    I1104 11:59:10.987620  1 controller.go:830] volume "pvc-8eed96e4-c157-11e7-8910-42010a840164" for claim "default/demo-snap-vol-claim" saved 
+    I1104 11:59:10.987740  1 controller.go:866] volume "pvc-8eed96e4-c157-11e7-8910-42010a840164" provisioned for claim "default/demo-snap-vol-claim"
+
+Now, let’s mount the “demo-snap-vol-claim” PersistentVolumeClaim onto a busybox-snapshot Pod to check whether the snapshot was restored properly. After the busybox-snapshot pod is in a running state, we can check the integrity of the files that were created before taking the snapshot.
+
+    apiVersion: v1
+    kind: Pod
+    metadata:
+      name: busybox-snapshot
+      namespace: default
+    spec:
+      containers:
+      - command:
+           - sh
+           - -c
+           - 'tail -f /dev/null'
+        image: busybox
+        imagePullPolicy: Always
+        name: busybox
+        volumeMounts:
+        - mountPath: /mnt/store2
+          name: demo-snap-vol
+      volumes:
+      - name: demo-snap-vol
+        persistentVolumeClaim:
+          claimName: demo-snap-vol-claim
+
+**Clean-Up: **We can delete the VolumeSnapshot resource, which will also delete the corresponding VolumeSnapshotData resource from K8s. This will not affect any PersistentVolumeClaims or PersistentVolumes we have already provisioned using the snapshot. Conversely, deleting any PersistentVolumeClaims or PersistentVolumes that have been used to create the snapshot or have been provisioned using a snapshot will not delete the snapshot from the OpenEBS backend. As such, we must delete them manually.
```

Comment:
```suggestion
**Clean-Up:** We can delete the VolumeSnapshot resource, which will also delete the corresponding VolumeSnapshotData resource from K8s. This will not affect any PersistentVolumeClaims or PersistentVolumes we have already provisioned using the snapshot. Conversely, deleting any PersistentVolumeClaims or PersistentVolumes that have been used to create the snapshot or have been provisioned using a snapshot will not delete the snapshot from the OpenEBS backend. As such, we must delete them manually.
```